utils.go

```diff
@@ -165,14 +198,23 @@ func SelectHopHints(ctx context.Context, lnd *lndclient.LndServices,
 			break
 		}
 
-		// Skip the channel if we already selected it.
+		// Channels of private nodes, inacitve, or public channels or
+		// those that have already been selected can be skipped in this
+		// iteration.
```

Comment:
Are we obeying the set of peers in `includeNodes` here? Seems like in this second pass we could end up adding channels with peers that we didn't intend to provide hints for? Either need to add them to skipcache or re-check here, I think? 

Test case for `includeNodes` adherence would be good as well, since restricting loop in to a subset of peers is an important feature for lots of people.